Ice balance attempt 3

mikepmanning opened the pull request:
All stuff for ice curing

Adds new affs and replaces necessary affs with overhaul stuff.

mikepmanning committed:
Adding wound ice entries to dict

mikepmanning committed:
adding in ice balance stuff

mikepmanning committed:
last of ice balance additions

Still need to add wounding info in

mikepmanning committed:
Ice balance stuff

mikepmanning committed:
More ice wounding stuff

mikepmanning committed:
pureblade lines added and additional ice stuff

mikepmanning committed:
Ice triggers added

Most Pureblade triggers added
Added 'apply ice to <bodypart> wounds to specify wound curing

mikepmanning committed:
adding in bleed-enhancer affs

mikepmanning committed:
Ice additions and bugfixes

Fixed some bugs and added more ice trigger lines

PB should be mostly set - may need to add in check for bleeder
combatstyle

mikepmanning committed:
Added anorexia fix for lucidity

added a fix for lucidity to react to anorexia symptom line

mikepmanning committed:
Adding stuff

sparkleherb fix,
Def line changes

mikepmanning committed:
BC triggers added

mikepmanning committed:
More trigger lines

Added Bruising functionality

Added in lots of trigger lines

Added in new combatstyle defs

Fixed deathsight def to use discernment skill

mikepmanning committed:
adjusted initial prios

initial ice prios adjusted

mikepmanning committed:
added general overhaul stuff

added overhaul replacement for
anorexia/asthma/slickness/damaged/mangled/amputatedlimbs and
fracturedskull

mikepmanning committed:
Final ice stuff added

mikepmanning committed:
Ice triggers and scripts

mikepmanning committed:
More ice stuff

clean up

mikepmanning committed:
final ice stuff

Removed arena requirement
added wounds to overhaul affs

mikepmanning committed:
Added better functionality

Added functionality to allow reversion to pre-warrior overhaul wound
system in case of reversion

vadi2 commented:
Looks like there's a merge conflict you need to resolve in the branch first.

mikepmanning commented:
How do I see what the conflicts are?

vadi2 committed:
Merge remote-tracking branch 'origin/Ice-Balance-attempt-3' into master-2

Conflicts:
	raw-mm.defs.lua

vadi2 commented:
I use SmartGit, so I clicked on the "command line instructions" - and it's Github giving command line instructions for merging the Ice-Balance-attempt-3 into your master branch. So I did that using the SmartGit UI instead, and there was a conflict where both branches modified some of the same lines in defs.lua. I set the changes from Ice-Balance-attempt-3 to override them.

Going to see if I can submit a pull request to your fork with the merge conflicts resolved that should update then your original pull request.

vadi2 commented:
Okay it's https://github.com/siddister/m-mf/pull/3, go and approve that one.

You modified the waterbreathing lines and submitted a PR for that into m-mf/master and then modified them separately in the ice balance branch without pulling in your changes from master - so you've made 2 separate edits to the same stuff, hence the conflict.

mikepmanning committed:
Merge pull request #3 from vadi2/master

Fixed merge conflict

mikepmanning commented:
Alright done, my bad there.

vadi2 commented:
All good. Some fixes needed -

^The salve mends your (head|chest|gut|left arm|left leg|right arm|right leg), leaving it with (light|heavy|critical) wounds.$

add a \ before a real . at the end on a few regex triggers.

vadi2 commented:
<regex>^mmsp convert$</regex>

Toggles should also provide a way of not toggling but explicitly setting an option, look at one of the current toggles for an example on how to do that.

vadi2 commented:
Remember to pull your branch before you make changes so you get my merge conflict resolution in.

vadi2 commented on raw-mm.aliases.lua:
```diff
@@ -458,13 +458,8 @@ function ashow()
-  echofn("Arena mode:       ")
-  setFgColor(unpack(getDefaultColorNums))
-  setUnderline(true)
-  echoLink(conf.arena and "enabled" or "disabled", "$(sys).tntf_set('arena', "..(conf.arena and "false" or "true").. ', false); svo.ashow()', (conf.arena and "Disable" or "Enable")..' arena triggers', true)
```
How come you got rid of showing arena mode?

vadi2 commented:
Any reason valid.oldwarrior isn't a conf option?

vadi2 commented:
In bruising.isadvisable, you might want to be checking doingaction("bleeding") as well... otherwise I think both bleeding and bruising will be sending clot commands and it could snowball.

vadi2 commented:
The old warrior toggle isn't used everywhere, ie:

```
  oncompleted = function ()
    removeaff("collapsedlungs")
    --removeaff("paregenchest")
    --addaff(dict.puncturedlung)
  end,
```

vadi2 commented:
if $(sys)["version"] >= 4 then looks dodgy... did you mean to use the oldwarrior check?

vadi2 commented on raw-mm.sp.lua:
```diff
@@ -123,16 +145,28 @@ sps.install = {
         t[#t+1] = 'mm.sp.setstancelevel("'..name..'", false, true)'
-        for amount = 275, 2000, 275 do
-          t[#t+1] = 'mm.sp.setstancelevel("'..name..'", '..amount..', true)'
+        if $(sys)["version"] >= 4 then
```
Looks dodgy... did you mean to use the oldwarrior check?

mikepmanning commented:
I'll take care of all this when I get home - most of it should be easy. I was rushing to get things done so people would have working systems with the release

vadi2 commented:
Okay do the fixes in another PR, I'll merge and release this.

vadi2 commented:
Okay I've created the release (https://github.com/m-mf/m-mf/releases/tag/4) - no time to add a detailed changelog for all the stuff. Could you write a changelog I should enter and make all the announcements?

mikepmanning commented:
Yeah, want me to post to forums too?

vadi2 commented:
Yeah - forums, m&m clan, etc.

Mention the new toggle alias you've got as well.

mikepmanning commented:
Changelog - version 4
- fixed generic def lines (mercy, kingdom, beauty, perfection, acquisitio, waterbreathing) 
- added ice balance and overhaul afflictions associated with warrior overhaul
- added bruising
- added stance/parry conversion
- combatstyle defs have been added for all the specs
- added 'mmsp convert' alias to switch between old warrior system and new warrior system in case of reversion
- added in overhaul cures for anorexia, slickness and asthma
- added in overhaul damagedlimbs and mutilatedlimbs to replace crippledlimbs and mangled/missinglimbs
- added damagedskull to replace fracturedskull with overhaul
- adjusted lucidity to work with anorexia so it can cure it
- damagedthroat blocks lucidity (may need to have it block other sipping)

vadi2 commented:
Changelog updated.

vadi2 commented:
I don't think you've added default priorities for ice by the way - the
bin/default_prios file doesn't list them.

On Wed, Jan 20, 2016 at 6:50 AM siddister notifications@github.com wrote:

> Changelog - version 4
> - fixed generic def lines (mercy, kingdom, beauty, perfection,
>   acquisitio, waterbreathing)
> - added ice balance and overhaul afflictions associated with warrior
>   overhaul -added bruising -added stance/parry conversion -combatstyle defs
>   have been added for all the specs -added 'mmsp convert' alias to switch
>   between old warrior system and new warrior system in case of reversion
>   -added in overhaul cures for anorexia, slickness and asthma -added in
>   overhaul damagedlimbs and mutilatedlimbs to replace crippledlimbs and
>   mangled/missinglimbs
>   - added damagedskull to replace fracturedskull with overhaul
>   - adjusted lucidity to work with anorexia so it can cure it
>   - damagedthroat blocks lucidity (may need to have it block other
>     sipping)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/m-mf/m-mf/pull/25#issuecomment-172982428.
